File: app/Models/Business.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Business extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = ['name', 'type', 'region', 'phone', 'email', 'logo', 'plan', 'status', 'vat_rate', 'currency', 'language', 'trial_ends_at', 'owner_name', 'address', 'tax_number', 'registration_number', 'website', 'social_media', 'dark_mode', 'settings'];

    protected $casts = ['trial_ends_at' => 'datetime', 'social_media' => 'array', 'dark_mode' => 'boolean', 'settings' => 'array'];

    public function branches()
    {
        return $this->hasMany(Branch::class);
    }

    public function users()
    {
        return $this->hasMany(User::class);
    }

    public function categories()
    {
        return $this->hasMany(Category::class);
    }

    public function products()
    {
        return $this->hasMany(Product::class);
    }

    public function customers()
    {
        return $this->hasMany(Customer::class);
    }

    public function suppliers()
    {
        return $this->hasMany(Supplier::class);
    }

    public function sales()
    {
        return $this->hasMany(Sale::class);
    }

    public function expenses()
    {
        return $this->hasMany(Expense::class);
    }

    public function subscriptions()
    {
        return $this->hasMany(Subscription::class);
    }

    public function stockTransfers()
    {
        return $this->hasMany(StockTransfer::class);
    }

    public function incomes()
    {
        return $this->hasMany(Income::class);
    }

    public function payments()
    {
        return $this->hasMany(Payment::class);
    }

    public function quotations()
    {
        return $this->hasMany(Quotation::class);
    }

    public function accounts()
    {
        return $this->hasMany(Account::class);
    }

    public function cashFlows()
    {
        return $this->hasMany(CashFlow::class);
    }

    public function reminders()
    {
        return $this->hasMany(Reminder::class);
    }

    public function greetingCards()
    {
        return $this->hasMany(GreetingCard::class);
    }

    public function businessCards()
    {
        return $this->hasMany(BusinessCard::class);
    }

    public function printerSetting()
    {
        return $this->hasOne(PrinterSetting::class);
    }

    public function notes()
    {
        return $this->hasMany(Note::class);
    }

    public function bills()
    {
        return $this->hasMany(Bill::class);
    }
}

File: mobile/lib/models/business.dart
class Business {
  final int id;
  final String name;
  final String? type;
  final String? region;
  final String? logo;
  final String? phone;
  final String? email;
  final String? address;
  final String? currency;
  final bool darkMode;
  final String status;

  Business({
    required this.id,
    required this.name,
    this.type,
    this.region,
    this.logo,
    this.phone,
    this.email,
    this.address,
    this.currency = 'TZS',
    this.darkMode = false,
    this.status = 'active',
  });

  factory Business.fromJson(Map<String, dynamic> json) {
    return Business(
      id: json['id'] as int,
      name: json['name'] as String,
      type: json['type'] as String?,
      region: json['region'] as String?,
      logo: json['logo'] as String?,
      phone: json['phone'] as String?,
      email: json['email'] as String?,
      address: json['address'] as String?,
      currency: json['currency'] as String? ?? 'TZS',
      darkMode: json['dark_mode'] as bool? ?? false,
      status: json['status'] as String? ?? 'active',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'type': type,
      'region': region,
      'logo': logo,
      'phone': phone,
      'email': email,
      'address': address,
      'currency': currency,
      'dark_mode': darkMode,
      'status': status,
    };
  }
}
